### ISSUE TITLE
Create training data and fine-tune model to extract Ex. 21 tables

### ISSUE THREAD

**Issue by katie-lamb** [catalyst-cooperative/mozilla-sec-eia#9]
I was able to extract a couple Ex. 21 tables with the Hugging Face Table Transformer, but in order to have this model generalize to all Ex. 21 tables we need to fine-tune the model with a small training set. It seems like the hardest part of the fine-tuning is figuring out how to create input training data in the correct format.

4/4 Update:
* Fine-tuning the Table Transformer model was hard, pivoting to trying to tune LayoutLM. It seems hard to figure out how to fine-tune TableTransformer, even though initial results with TableTransformer are much better
* [FUNSD dataset](https://huggingface.co/datasets/nielsr/funsd-layoutlmv3) is what Katie is trying to emulate with training data labeling because it can be fed into the LayoutLM model to fine-tune. This requires the following columns: ID, tokens (words), bounding boxes, named entity recognition tags (which we are defining), and image
* Is it the right call to try to make [LayoutLM](https://huggingface.co/docs/transformers/model_doc/layoutlmv3) work? We could spend a bunch of time making training data that works for LayoutLM and then need to pivot models. We don’t want to recreate the training data from scratch in that case, so how do we make the training data generalizable? Want this data to work on a simpler named entity recognition model
* What is the fastest way to create this training data? Using the Label Studio tool on images or going simpler and just using the coordinates from PDFs + OCR? Trenton has a script that can go from text-based PDF to image with bounding boxes
* Blog posts that I’m referencing: [using label studio to fine-tune LayoutLM](https://medium.com/@ravisatvik.192/unleashing-the-power-of-layoutlm-extracting-entities-from-structured-documents-made-easy-5d82c6290ec7), [using commercial tool to fine-tune LayoutLM](https://medium.com/p/1e55869336d4) (and there are 2 follow up posts too with later versions of LayoutLM)

```[tasklist]
### Tasks
- [x] Read through fine-tuning tutorial and look at the Training class to see how training data should be formatted
- [ ] #15
- [x] Label ~50 training examples
- [x] Fine-tune the LayoutLM model with a training set
- [x] Validate the fine-tuned model on a test set
- [x] Create the bones of a LayoutLM modeling module - moved to separate issue.
```

**Comment by jdangerx** [comment]
@katie-lamb sounds like you're going down the LayoutLM path - how is that going? Are you running into any issues?

**Comment by katie-lamb** [comment]
@jdangerx No issues yet! It's been slow going trying to get Label Studio to work as a labeling UI but I haven't hit any major roadblocks. I was able to get the fine-tuning working on a dataset of 2 docs so I know the format is correct.
